src/main/java/org/aksw/simba/lemming/metrics/single/MaxVertexDegreeMetric.java

```diff
@@ -29,4 +34,181 @@ public double apply(ColouredGraph graph) {
+    public UpdatableMetricResult update(TripleBaseSingleID triple, ColouredGraph graph, boolean graphOperation,
+            UpdatableMetricResult previousResult, VertexDegrees mVertexDegrees) {
+        UpdatableMetricResult newMetricResult;
+
+        if (direction == DIRECTION.in) {
+
+            if (graphOperation) { // graphOperation is true then add an edge otherwise its remove an edge
+                newMetricResult = metricComputationMaxDegree(graph, DIRECTION.in,
+                        triple.headId, triple, 1, previousResult, mVertexDegrees);
+            } else {
+                newMetricResult = metricComputationMaxDegree(graph, DIRECTION.in,
+                        triple.headId, triple, -1, previousResult, mVertexDegrees);
+            }
+
+        } else {
+            if (graphOperation) {
+                newMetricResult = metricComputationMaxDegree(graph, DIRECTION.out,
+                        triple.tailId, triple, 1, previousResult, mVertexDegrees);
+            } else {
+                newMetricResult = metricComputationMaxDegree(graph, DIRECTION.out,
+                        triple.tailId, triple, -1, previousResult, mVertexDegrees);
+            }
+
+        }
+
+        return newMetricResult;
+    }
+
+    /**
+     * The method contains logic that reduces the number of calls to apply method
+     * for the max vertex degree metric.
+     * 
+     * @param metric
+     *            - metric which should be calculated.
+     * @param graph
+     *            - input graph.
+     * @param metricName
+     *            - can be "RemoveAnEdge" or "AddAnEdge" indicating how the edge is
+     *            modified.
+     * @param direction
+     *            - this is in or out based on the operation.
+     * @param vertexID
+     *            - The vertex that is modified.
+     * @return
+     */
+    private UpdatableMetricResult metricComputationMaxDegree(ColouredGraph graph, 
+            DIRECTION direction, int vertexID, TripleBaseSingleID triple, int updateVertexDegree,
+            UpdatableMetricResult previousResult, VertexDegrees mVertexDegrees) {
+        double metVal;
+
+        MaxVertexDegreeMetricResult metricResultTempObj = new MaxVertexDegreeMetricResult(getName(), 0.0);
+
+        if (previousResult instanceof MaxVertexDegreeMetricResult) {
+            // Set previously stored maps
+            metricResultTempObj
+                    .setCandidatesMetricSet(((MaxVertexDegreeMetricResult) previousResult).getmMapCandidatesMetric());
+            metricResultTempObj.setMaxVertexDegree(
+                    ((MaxVertexDegreeMetricResult) previousResult).getMaxVertexDegree());
+        }
+
+        IntSet intSetTemp = metricResultTempObj.getmMapCandidatesMetric();
+        // Get the current candidate set
+
+        if (intSetTemp.size() == 0) { // Initially the Candidate set will be empty, hence need to call the apply
+                                      // method and store the candidates
```

Comment:
But why we are here if there is no initial result that can be updated? Shouldn't we have a different method for that? :confused:

Reply:
Done. This logic is removed now. Max vertex degree metric makes use of number of vertices having maximum degree and checks if the metric should be updated or not.